src/java.net.http/share/classes/jdk/internal/net/http/Http2ClientImpl.java

```diff
@@ -153,14 +152,19 @@
      */
     boolean offerConnection(Http2Connection c) {
         if (debug.on()) debug.log("offering to the connection pool: %s", c);
-        if (c.closed || c.finalStream()) {
+        if (!c.isOpen() || c.finalStream()) {
```

Comment:
Is this check for isOpen() not redundant given the same check added inside the synchronized block?

Reply:
Possibly - but it allows to break out fast without having to enter the synchronized block.

Reply:
Okay, that's fine.